### PR TITLE
Fix OCI trim error

### PR DESCRIPTION
**Pull request by Arkeins**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  |  #19327 

The error occurs with PHP 8.1

### CODE REVIEW

**Comment by WinterSilence** [comment]
redundant changes - `trim()` and `stripos()` convert argument to string

**Comment by Arkeins** [comment]
The first argument is changed as a string, not the function return.

This ensure that the first argument is not `null`, which cause the error.

**Comment by terabytesoftw** [comment]
> The first argument is changed as a string, not the function return.
> 
> This ensure that the first argument is not `null`, which cause the error (using PHP 8.1).

why not ?

```php
if ($column['DATA_DEFAULT'] !== null && stripos($column['DATA_DEFAULT'], 'timestamp') !== false) {
```

**Comment by Arkeins** [comment]
> > The first argument is changed as a string, not the function return.
> > This ensure that the first argument is not `null`, which cause the error (using PHP 8.1).
> 
> why not ?
> 
> ```
> if ($column['DATA_DEFAULT'] !== null && stripos($column['DATA_DEFAULT'], 'timestamp') !== false) {
> ```

Unless I am mistaken, it does the same thing ?

This is my first time contributing, sorry if I am doing this wrong ^^'

**Comment by terabytesoftw** [comment]
> > > The first argument is changed as a string, not the function return.
> > > This ensure that the first argument is not , which cause the error (using PHP 8.1).`null`
> > 
> > 
> > why not ?
> > ```
> > if ($column['DATA_DEFAULT'] !== null && stripos($column['DATA_DEFAULT'], 'timestamp') !== false) {
> > ```
> 
> Unless I am mistaken, it does the same thing ?
> 
> This is my first time contributing, sorry if I am doing this wrong ^^'

Thanks for contributing, your solution is fine, they do the same, but it is more human readable, see what you want to validate, which is that the first argument is not `null`.

**Comment by Arkeins** [comment]
I used the `(string)` because the aim of this fix is to fix the behavior of PHP, not the logic of the function, and that adding it to the `if` make it heavier to read, if that make sens.

In the same idea, should i change:

```php
trim((string) $length) === '' ? null : (int) $length
```

to:

```php
$length !== null && trim($length) === '' ? null : (int) $length
```

?

File: framework/CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Yii Framework 2 Change Log
 - Enh #19304: Add filtering validator `yii\validators\TrimValidator` (WinterSilence)
 - Enh #19309: Optimize `yii\base\Model::attributes()` (WinterSilence)
 - Bug #19322: Revert force setting value to empty string in case it's `null` in `yii\validators\FilterValidator::validateAttribute()` (bizley)
+- Bug # : Passing null to parameter #1 ($string) of type string is deprecated in `yii\db\oci\Schema` (Arkeins)
```
Comment (samdark) [inline framework/CHANGELOG.md:14]:
```suggestion
- Bug #19328: Passing null to parameter #1 ($string) of type string is deprecated in `yii\db\oci\Schema` (Arkeins)
```

**Review by samdark — APPROVED** [review]

**Review by Arhell — COMMENTED** [review]
Please resolve conflicting file

**Comment by Arkeins** [comment]
> Please resolve conflicting file

Done!

**Comment by samdark** [comment]
Thanks.

**Pull request merged**